### ISSUE TITLE
Ziglua 1.0.0

### ISSUE THREAD

**Issue by natecraddock** [natecraddock/ziglua#32]
This lists the changes I want to make before releasing a 1.0.0. Subtasks are used for more details.

At a high level, I want Ziglua to be in a state that is more easily maintained, and generally be feature complete enough to reflect what future versions would include

**Comment by VisenDev** [comment]
@natecraddock Regarding issue #31 I think the best course of action would be to wait for #52 to be completed to avoid duplicate work

Then I can rework the implementation from [my current fork](https://github.com/VisenDev/ziglua) to use the new lib.zig file (and implement any requested changes) then send over a pr? Assuming you still had positive feelings about providing that functionality

**Comment by natecraddock** [comment]
@VisenDev I agree. I should have the lib changes finished within the next week or so if everything goes well.

And the more I think about it, the more I like the ideas in #31. I'm not sure how the final implementation will look, but Ziglua should absolutely support some level of convenience/meta functions that take advantage of comptime

**Comment by zoriya** [comment]
Hello, just a lurker that started using zig and ziglua recently and wanted to thank the both of you for this library and the promising convenience it will offer.

**Comment by natecraddock** [comment]
@zoriya it is always nice to hear a thank you, I appreciate you taking the time to say that!

If you need help or have ideas for improvement as you use ziglua please don't hesitate to share them!

**Comment by mischief** [comment]
> I wonder if [#75](https://github.com/natecraddock/ziglua/issues/75) is actually resolved by [#157](https://github.com/natecraddock/ziglua/pull/157) ?

i don't think so. that seems to set the name of the build of the vendored lua. unclear how that would help use the system library and suppress building the vendored one.

**Comment by robbielyman** [comment]
ah, i see. you are right.

i'd definitely like to see #75 supported, personally!
